src/common.jl

```diff
@@ -0,0 +1,74 @@
+struct TestEnvError <: Exception
+    msg::AbstractString
+end
+
+function Base.showerror(io::IO, ex::TestEnvError, bt; backtrace=true)
+    printstyled(io, ex.msg, color=Base.error_color())
+end
+
+
+current_pkg_name() = Context().env.pkg.name
+
+"""
+   ctx, pkgspec = ctx_and_pkgspec(pkg::AbstractString)
+   
+For a given package name `pkg`, instantiate a `Context` for it, and return that `Context`,
+and it's `PackageSpec`.
+"""
+function ctx_and_pkgspec(pkg::AbstractString)
+    pkgspec = deepcopy(PackageSpec(pkg))
+    ctx = Context()
+    isinstalled!(ctx, pkgspec) || throw(TestEnvError("$pkg not installed 👻"))
+    Pkg.instantiate(ctx)
+    return ctx, pkgspec
+end
+
+"""
+    isinstalled!(ctx::Context, pkgspec::Pkg.Types.PackageSpec)
+
+Checks if the package is installed by using `ensure_resolved` from `Pkg/src/Types.jl`.
+This function fails if the package is not installed, but here we wrap it in a
+try-catch as we may want to test another package after the one that isn't installed.
+"""
+function isinstalled!(ctx::Context, pkgspec::Pkg.Types.PackageSpec)
+    project_resolve!(ctx.env, [pkgspec])
+    project_deps_resolve!(ctx.env, [pkgspec])
+    manifest_resolve!(ctx.env, [pkgspec])
+    try
+        ensure_resolved(ctx.env, [pkgspec])
+    catch err
+        err isa MethodError && rethrow()
+        return false
+    end
+    return true
+end
+
+function test_dir_has_project_file(ctx, pkgspec)
+    return isfile(joinpath(get_test_dir(ctx, pkgspec), "Project.toml"))
+end
+
+"""
+    get_test_dir(ctx::Context, pkgspec::Pkg.Types.PackageSpec)
+
+Gets the testfile path of the package. Code for each Julia version mirrors that found 
+in `Pkg/src/Operations.jl`.
+"""
+function get_test_dir(ctx::Context, pkgspec::Pkg.Types.PackageSpec)
+    pkgspec.special_action = Pkg.Types.PKGSPEC_TESTED
+    if is_project_uuid(ctx.env, pkgspec.uuid)
+        pkgspec.version = ctx.env.pkg.version
+        pkgfilepath = dirname(ctx.env.project_file)
+    else        
+        info = manifest_info(ctx.env, pkgspec.uuid)
+        if haskey(info, "git-tree-sha1")
+            pkgfilepath = find_installed(pkgspec.name, pkgspec.uuid, SHA1(info["git-tree-sha1"]))
+        elseif haskey(info, "path")
+            pkgfilepath =  project_rel_path(ctx, info["path"])
+        elseif pkgspec.uuid in keys(ctx.stdlibs)
+            pkgfilepath = Pkg.Types.stdlib_path(pkgspec.name)
+        else
+            throw(TestEnvError("Could not find either `git-tree-sha1` or `path` for package $(pkgspec.name)"))
+        end
+    end
+    return joinpath(pkgfilepath, "test")
```

Comment:
Could delete this line as per `v1.1` branch

Reply:
If we don't have it the function doesn't do what it says it does.
Not that we use this function's actual effect in v1.0-1.2, we only use it's side-effect